webapp/views/App/views/Designer/TaxonomyList/Row/store/useTaxonomyRow.js

```diff
@@ -0,0 +1,18 @@
+import { useState, useEffect } from 'react'
+
+import { useActions } from './actions'
+import { State } from './state'
+
+export const useTaxonomyRow = (props) => {
+  const { selectedItemUuid } = props
+
+  const [state, setState] = useState(State.create(props))
```

Comment:
function State.create(props) will be invoked at each re-render. 
to use a function in the useState initialization you need to pass a function.
```  
const [state, setState] = useState(() => State.create(props)) 
```

Reply:
is this the right place of should use this creator into the component?